src/js/controllers/mapController.ts

```diff
@@ -1006,6 +1007,45 @@ export class MapController {
     }
   }
 
+  async getDocuments(
+    urlProperties: URLProperties
+  ): Promise<Array<Attachment> | null> {
+    const { sublayerID, specificFeatureID, layerID } = urlProperties;
+    let endPoint = '';
+    const layer = this._map?.findLayerById(layerID);
+
+    if ((layer as any).sublayers.length && specificFeatureID) {
+      const sublayer = (layer as any).sublayers.items.filter(
```

Comment:
let's move this function to some helper file instead of mapController. It is getting really big already.

We can just create a small function in mapController that accepts a layer ID and returns a layer it is found, but whole logic can live elsewhere.